addons/mail/static/src/components/attachment_box/attachment_box.xml

```diff
@@ -19,21 +19,21 @@
                             t-ref="dropzone"
                         />
                     </t>
-                    <t t-if="chatter.attachmentList">
+                    <t t-if="attachmentBoxView.chatter.attachmentList">
                         <AttachmentList
                             class="o_attachmentBox_attachmentList"
-                            attachmentListLocalId="chatter.attachmentList.localId"
-                            t-on-o-attachment-removed="_onAttachmentRemoved"
+                            attachmentListLocalId="attachmentBoxView.chatter.attachmentList.localId"
+                            t-on-o-attachment-removed="attachmentBoxView.onAttachmentRemoved"
```

Comment:
Remove this event, AttachmentList can now be aware of its `chatter` -> `attachmentBoxView` in models and directly make the appropriate handling by itself in that case (instead of triggering `o-attachment-removed`, it could do the code of `onAttachmentRemoved` directly instead (which is in that case triggering `o-attachments-changed`).

Reply:
Looking at this, there is no handler for `o-attachments-changed`